specs/experimental/gov-delegation.md

```diff
@@ -0,0 +1,440 @@
+and `_delegatee` addresses. Afterwards, the function MUST call the [`_delegate`](#_delegate) function with the `_delegator`
+and basic delegation parameters.
+
+At the end, the `delegateFromToken` function MUST emit a `DelegationCreated` event with the given function parameters.
+
+#### `delegateBatched`
+
+Delegates voting power to multiple addresses (delegatees) using the delegation array. This function is intended to be
+called by users.
+
+```solidity
+function delegateBatched(Delegation[] calldata _delegations) external
+```
+
+This function MUST enforce the migration logic, as specified in the [Migration](#migration) section, for `msg.sender`
+and every delegatee address in the `_delegations` array. Afterwards, the function MUST call the [`_delegate`](#_delegate)
+function with the `msg.sender` and `_delegations` parameters.
+
+At the end, the `delegateBatched` function MUST emit a `DelegationCreated` event with the given function parameters.
+
+#### `afterTokenTransfer`
+
+Updates the voting power of two addresses (`_from` and `_to`) after a token transfer. This function MUST only be callable
+by the `GovernanceToken` contract.
+
+```solidity
+function afterTokenTransfer(address _from, address _to, uint256 _amount) external
+```
+
+This function MUST enforce the migration logic, as specified in the [Migration](#migration) section, for the `_from`
+and `_to` addresses.
+
+If the token transfer is due to a mint or burn operation, the function MUST update the total supply checkpoints with
+the new total supply (subtracting `_amount` for a burn operation and adding `_amount` for a mint operation). Afterwards,
+the function MUST calculate and update the voting power of the delegations of `_from` and `_to`. This step MUST apply
+the voting power adjustments to the checkpoints of the `_from`, `_to`, and delegatee addresses.
+
+#### `migrateAccounts`
+
+Migrates the delegation state of the given accounts from the `GovernanceToken` to the `GovernanceDelegation` contract. This
+function MUST iterate over the list of `_accounts` addresses and apply the logic specified in the
+[Migration](#migration) section.
+
+```solidity
+function migrateAccounts(address[] calldata _accounts) external
+```
+
+#### `_delegate`
+
+Delegates voting power from the `_delegator` to the specified delegates in `_newDelegations`. This function MUST
+override any existing delegations of `_delegator`.
+
+```solidity
+function _delegate(address _delegator, Delegation[] memory _newDelegations) internal virtual
+```
+
+The `_delegate` function MUST first check that the length of `_newDelegations` does not exceed `MAX_DELEGATIONS`.
+Then, it MUST calculate the voting power of all active delegations of the `_delegator`, and calculate the voting
+power adjustments for the new delegatee set using `_newDelegations`. Afterwards, the function MUST update
+the voting power of the current and new set of delegatees. Throughout this process, the function MUST use the
```

Comment:
Use of `DelegationAdjustment` and `Op` here seems like an implementation detail and can probably be removed, unless that's part of the public interface. From what I can see this isn't exposed publicly and therefore can be removed.